### ISSUE TITLE
Expanded Lights crashes (using updated Plib)

### ISSUE THREAD

**Issue by MegaJani** [skairunner/sky-oni-mods#111]
Relevant error in the log:

[23:45:04.728] [1] [ERROR] [PLib/ExpandedLights] System.Reflection.AmbiguousMatchException Ambiguous match found.   at System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) [0x00059] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Type.GetMethod (System.String name, System.Reflection.BindingFlags bindingAttr) [0x0000e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at PeterHan.PLib.Core.ExtensionMethods.Patch (HarmonyLib.Harmony instance, System.Type type, System.String methodName, HarmonyLib.HarmonyMethod prefix, HarmonyLib.HarmonyMethod postfix) [0x00028] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 
[23:45:04.772] [1] [INFO] Expanded Lights's assembly declared the method PeterHan.PLib.Core.PUtil:LogException in the stack trace, adding to referenced mods list
[23:45:04.773] [1] [INFO] Mod Updater's assembly declared the method PeterHan.PLib.Core.PRemoteComponent:DoInitialize in the stack trace, adding to referenced mods list
[PLib/ExpandedLights] System.Reflection.AmbiguousMatchException Ambiguous match found.   at System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) [0x00059] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Type.GetMethod (System.String name, System.Reflection.BindingFlags bindingAttr) [0x0000e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at PeterHan.PLib.Core.ExtensionMethods.Patch (HarmonyLib.Harmony instance, System.Type type, System.String methodName, HarmonyLib.HarmonyMethod prefix, HarmonyLib.HarmonyMethod postfix) [0x00028] in <560e3342ab0c4c34ba39f2a0917dbd78>:0

**Comment by fruixs** [comment]
same
[Player-prev.log](https://github.com/user-attachments/files/16357164/Player-prev.log)

```
[PLib/ExpandedLights] System.Reflection.AmbiguousMatchException Ambiguous match found.   at System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) [0x00059] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Type.GetMethod (System.String name, System.Reflection.BindingFlags bindingAttr) [0x0000e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at PeterHan.PLib.Core.ExtensionMethods.Patch (HarmonyLib.Harmony instance, System.Type type, System.String methodName, HarmonyLib.HarmonyMethod prefix, HarmonyLib.HarmonyMethod postfix) [0x00028] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 

  at UnityEngine.Debug.LogErrorFormat (System.String format, System.Object[] args) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Debug.LogErrorFormat (System.String format, System.Object[] args) [0x00000] in <3cae53dcd334432a8d9641184e90df7c>:0 
  at PeterHan.PLib.Core.PUtil.LogException (System.Exception thrown) [0x00000] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 
  at PeterHan.PLib.Core.ExtensionMethods.Patch (HarmonyLib.Harmony instance, System.Type type, System.String methodName, HarmonyLib.HarmonyMethod prefix, HarmonyLib.HarmonyMethod postfix) [0x00000] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 
  at PeterHan.PLib.Lighting.LightingPatches.ApplyPatches (HarmonyLib.Harmony plibInstance) [0x00000] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 
  at PeterHan.PLib.Lighting.PLightManager.Initialize (HarmonyLib.Harmony plibInstance) [0x00000] in <560e3342ab0c4c34ba39f2a0917dbd78>:0 
  at PeterHan.PLib.Core.PRemoteComponent.DoInitialize (HarmonyLib.Harmony plibInstance) [0x00000] in <3456cf41afb34e5ebbcb55b5776845b3>:0 
  at PeterHan.PLib.Core.PRegistryComponent.Instantiate () [0x00000] in <3456cf41afb34e5ebbcb55b5776845b3>:0 
  at PeterHan.PLib.Core.PRegistryComponent.ApplyLatest () [0x00000] in <3456cf41afb34e5ebbcb55b5776845b3>:0 
  at KMod.Mod.KMod.Mod.PostLoad_Patch1 (KMod.Mod , System.Collections.Generic.IReadOnlyList`1[T] ) [0x00000] in <ac8788b5d0fb4b89a20068ab3b6a25de>:0 
  at KMod.Manager.Load (KMod.Content content) [0x00000] in <ac8788b5d0fb4b89a20068ab3b6a25de>:0 
  at Global.Awake () [0x00000] in <ac8788b5d0fb4b89a20068ab3b6a25de>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent_Injected (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3& pos, UnityEngine.Quaternion& rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3 pos, UnityEngine.Quaternion rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate[T] (T original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.GameObject parent, System.String name, System.Boolean initialize_id, System.Int32 gameLayer) [0x00000] in <3cae53dcd334432a8d9641184e90df7c>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.GameObject parent, System.String name) [0x00000] in <3cae53dcd334432a8d9641184e90df7c>:0 
  at LaunchInitializer.Update () [0x00000] in <ac8788b5d0fb4b89a20068ab3b6a25de>:0 
Build: U52-623230-SC
```

**Comment by DarkRoku12** [comment]
Fix here:
https://github.com/skairunner/sky-oni-mods/pull/112
Steam mod:
https://steamcommunity.com/sharedfiles/filedetails/?id=3311098779

**Comment by MegaJani** [comment]
> Fix here: #112 Steam mod: https://steamcommunity.com/sharedfiles/filedetails/?id=3311098779

Bless you
